contracts/test/integration/IntegrationTestPriceBoundPSMDai.t.sol

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {ERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {MockPCVDepositV2} from "../../mock/MockPCVDepositV2.sol";
+import {IPCVDeposit} from "../../pcv/IPCVDeposit.sol";
+import {MockERC20} from "../../mock/MockERC20.sol";
+import {OraclePassThrough} from "../../oracle/OraclePassThrough.sol";
+import {ICore} from "../../core/ICore.sol";
+import {Core} from "../../core/Core.sol";
+import {IVolt, Volt} from "../../volt/Volt.sol";
+import {PriceBoundPSM, PegStabilityModule} from "../../peg/PriceBoundPSM.sol";
+import {getCore, getMainnetAddresses, VoltTestAddresses} from "../unit/utils/Fixtures.sol";
+import {ERC20CompoundPCVDeposit} from "../../pcv/compound/ERC20CompoundPCVDeposit.sol";
+import {Vm} from "./../unit/utils/Vm.sol";
+import {DSTest} from "./../unit/utils/DSTest.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {Constants} from "../../Constants.sol";
+
+contract IntegrationTestPriceBoundPSMDaiTest is DSTest {
+    using SafeCast for *;
+    PriceBoundPSM private psm;
+    ICore private core = ICore(MainnetAddresses.CORE);
+    IVolt private volt = IVolt(MainnetAddresses.VOLT);
+    IVolt private dai = IVolt(MainnetAddresses.DAI);
+    IVolt private underlyingToken = dai;
+
+    /// ------------ Minting and RateLimited System Params ------------
+
+    uint256 public constant mintAmount = 10_000_000e18;
+    uint256 public constant mintAmountPSM = 10_000_000e18;
+    uint256 public constant bufferCap = 10_000_000e18;
+    uint256 public constant individualMaxBufferCap = 5_000_000e18;
+    uint256 public constant rps = 10_000e18;
+
+    /// ------------ Oracle System Params ------------
+
+    /// @notice prices during test will increase 1% monthly
+    int256 public constant monthlyChangeRateBasisPoints = 100;
+    uint256 public constant maxDeviationThresholdBasisPoints = 1_000;
+
+    /// @notice Oracle Pass Through contract
+    OraclePassThrough public oracle =
+        OraclePassThrough(MainnetAddresses.ORACLE_PASS_THROUGH);
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint256 voltFloorPrice = 9_000; /// 1 volt for .9 dai is the max allowable price
+    uint256 voltCeilingPrice = 10_000; /// 1 volt for 1 dai is the minimum price
+
+    function setUp() public {
+        PegStabilityModule.OracleParams memory oracleParams;
+
+        oracleParams = PegStabilityModule.OracleParams({
+            coreAddress: address(core),
+            oracleAddress: address(oracle),
+            backupOracle: address(0),
+            decimalsNormalizer: 0,
+            doInvert: true
+        });
+
+        /// create PSM
+        psm = new PriceBoundPSM(
```

Comment:
This test should leverage the deployed DAI PSM and run the VIP-7 proposal simulation against mainnet. It should not deploy a new smart contract.

Reply:
Not sure if I agree, as these tests were written prior to the deployment of DAI PSM with the same configuration we would deploy the PSM to assert that it functions as we would expect it to once deployed. So functionally they're the same, so we could replace it with the deployed version, but this feels logically equivalent to deploying first and then running integration tests on a deployed contract. 

As for the simulation this VIP doesn't effect the DAI PSM in terms of being able to mint or redeem (it doesn't send or remove any tokens from it), I think a better place for a simulation on this part would be in VIP 8.

Reply:
The issue with your argument is it assumes parity between the deployed smart contracts and what happens in this test, which isn't the case because the DAI psm had different params, VoltSystemOracle instead of OraclePassThrough. Even if that weren't the case and you could assume parity between this test and the deployment script, you still want to do a simulation on mainnet against your actual contract once it is deployed to get the final ok that your system is working before having things go live.